src/pb_stub.cc

```diff
@@ -784,12 +809,32 @@ Stub::ProcessReturnedResponses(
             "return list, found type '" +
             std::string(py::str(py_responses[i].get_type())) + "'.");
       }
+
       std::shared_ptr<InferResponse> response =
           py_responses[i].cast<std::shared_ptr<InferResponse>>();
-      request->GetResponseSender()->Send(
-          response, TRITONSERVER_RESPONSE_COMPLETE_FINAL);
+      request->GetResponseSender()->UpdateStateAndCounters(response, TRITONSERVER_RESPONSE_COMPLETE_FINAL);
     }
   }
+  response_batch = std::move(shm_pool_->Construct<char>(sizeof(IPCMessageShm) + 
```

Comment:
```suggestion
  }
  // Initialize the 'response_batch' and send the returned response
  response_batch = std::move(shm_pool_->Construct<char>(sizeof(IPCMessageShm) + 
```
Maybe adding a comment on what will happen next?